src/bot.js

```diff
@@ -16,14 +16,15 @@ const retweet = () => {
     q: config.query,
     result_type: config.result_type,
     lang: config.lang,
-    tweet_mode: "extended"
+    tweet_mode: 'extended'
   };
 
   TwitterBot.get('search/tweets', params, (err, data) => {
     // when no errors
     if (!err) {
-      if (data.statuses[0].full_text.split('#').length - 1 === 1) {
+            if ((data.statuses[0].full_text.length <140 ) && (data.statuses[0].full_text.split('#').length - 1 === 1) && (data.statuses[0].full_text.toLowerCase().includes("#100daysofcode"))){
```

Comment:
```suggestion
      if ((data.statuses[0].entities.hashtags.length === 1) && (data.statuses[0].entities.hashtags[0].text.toLowerCase() === "100daysofcode")){
```

So I think we could take this approach to parse the hashtags, then.

Reply:
Thank you for suggesting this change. I have tested locally and the new logic is rejecting spam tweets very efficiently.

[Here is an example](https://twitter.com/noor_sudheer/status/1554423959118036992) of one of the successful retweets.

I'll continue to test today and update the PR in case all goes well. 😃

Reply:
I have been testing the new logic, but it seems the bot still is still retweeting some spam. Here is a tweet: https://twitter.com/worldwideRaay/status/1554209029311922176

In this case, `data.statuses[0].entities.hashtags.length` returned `1` instead of `5`.

I think, the issue still lies with the truncated text of `full_text`:

```bash

  full_text: 'RT @worldwideRaay: Day 20/100 of #100daysofcode\n' +
    '\n' +
    'Finished and understood the tricky challenge on loops, array and functions that I mentione…',

```

Reply:
Agreed and `extended` mode should work.  I have been trying to figure that out for a while now.

Reply:
I'm not sure if this library is doing things behind the scenes, but it looks like we should actually be seeing an `extended_tweet` property that has the extended values we want? https://developer.twitter.com/en/docs/twitter-api/premium/data-dictionary/overview

Reply:
I have been gathering the logs while making the changes, but I couldn't find an `extended_tweet` property. 
Once the `extended_tweet` mode is enabled, we are getting the tweet text in the `full_text` property instead of `text`.